src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderIndividualSchedule.kt

```diff
@@ -0,0 +1,108 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa
+
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.EnumType
+import jakarta.persistence.Enumerated
+import jakarta.persistence.FetchType
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.ManyToOne
+import jakarta.persistence.SequenceGenerator
+import jakarta.persistence.Table
+import org.hibernate.Hibernate
+import org.hibernate.annotations.JoinColumnOrFormula
+import org.hibernate.annotations.JoinColumnsOrFormulas
+import org.hibernate.annotations.JoinFormula
+import org.hibernate.annotations.NotFound
+import org.hibernate.annotations.NotFoundAction
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+enum class EventClass {
+  EXT_MOV, INT_MOV, COMM
+}
+
+@Entity
+@Table(name = "OFFENDER_IND_SCHEDULES")
+class OffenderIndividualSchedule(
+
+  @Id
+  @SequenceGenerator(name = "EVENT_ID", sequenceName = "EVENT_ID", allocationSize = 1)
+  @GeneratedValue(generator = "EVENT_ID")
+  @Column(name = "EVENT_ID", nullable = false)
+  val id: Long = 0,
+
+  @ManyToOne(optional = false, fetch = FetchType.LAZY)
+  @JoinColumn(name = "OFFENDER_BOOK_ID", nullable = false)
+  val offenderBooking: OffenderBooking,
+
+  @Column
+  val eventDate: LocalDate? = null,
+
+  @Column
+  val startTime: LocalDateTime? = null,
+
+  @Column
+  val endTime: LocalDateTime? = null,
+
+  @Enumerated(EnumType.STRING)
+  @Column(nullable = false)
+  val eventClass: EventClass = EventClass.INT_MOV, // INT_MOV only for now
+
+  @Column(nullable = false)
+  val eventType: String = "APP", // APP for appointment
+
+  @ManyToOne(optional = false)
+  @NotFound(action = NotFoundAction.EXCEPTION)
```

Comment:
maybe have a look at preprod and see if we ever have missing values in which case you could drop the NotFound

Reply:
👍

Reply:
Changed to ignore as there could be existing data with codes not in the ref list 🤷